Add USBSID-Pico hardware SID support

LouDnl opened the pull request:
This pull request adds USBSID-Pico hardware SID support.
USBSID-Pico also has official support in Vice, Jsidplay2 and Denise.

Please let me know if anything needs changing.

There is a known issue with dual and triple SID play, see https://github.com/LouDnl/libsidplayfp-usbsid/issues/5

drfiemost commented on .github/workflows/build.yml:
```diff
@@ -6,6 +6,9 @@ on:
   pull_request:
+    branches: [ master ]
+    # Allows you to run this workflow manually from the Actions tab when needed
+  workflow_dispatch:
```
What's this? Seems unrelated

drfiemost commented on .github/workflows/build.yml:
```diff
@@ -51,14 +55,16 @@ jobs:
           libgcrypt \
           xa \
+          libusb \
+          libusb-compat \
```
Is libusb-compat really required?

LouDnl commented:
It seems some changed not meant for the pull request got mixed up between switching workstations. Will fix.

LouDnl commented:
> A few notes:
> 
> * the branch have conflicts and can't be rebased and merged, please check  

Did not come across any merge conflicts when using latest master branch and merging in the pull-request, can you check please? See merge log below.

> * the pkg-config file [(libsidplayfp.pc.in)](https://github.com/libsidplayfp/libsidplayfp/blob/89d4f9d9f77ca68350a9fd8b72f0bdea5bbdb3f2/libsidplayfp.pc.in#L9) should export the libusb in the private section if the builder is enabled (look at how it is done for exSID)  

Fixed in [973fb58](https://github.com/libsidplayfp/libsidplayfp/pull/216/commits/973fb58e5924815c5e08ddd4fb45a0d3e2672d4a)

> * the usb lib should be added to builder specific flags, similar to how it's done for exSID: https://github.com/libsidplayfp/libsidplayfp/blob/89d4f9d9f77ca68350a9fd8b72f0bdea5bbdb3f2/Makefile.am#L388-L389  

Fixed in [f66e161](https://github.com/libsidplayfp/libsidplayfp/pull/216/commits/f66e161d0a7c147afbb8642c2ff70af0faedc204)

> * with the above, changes in `tests/Makefile.am` can be reverted  

Fixed in [bfa78c8](https://github.com/libsidplayfp/libsidplayfp/pull/216/commits/bfa78c82e1580dd82dc1c6164f9f91c7d2a9ff39)


Merge logs:
```shell
❯ git clone git@github.com:libsidplayfp/libsidplayfp libsidplayfp
Cloning into 'libsidplayfp'...
remote: Enumerating objects: 48040, done.
remote: Counting objects: 100% (5538/5538), done.
remote: Compressing objects: 100% (790/790), done.
remote: Total 48040 (delta 5244), reused 4780 (delta 4743), pack-reused 42502 (from 3)
Receiving objects: 100% (48040/48040), 34.70 MiB | 15.60 MiB/s, done.
Resolving deltas: 100% (41834/41834), done.
❯ cd $_
❯ git fetch origin pull/216/head:uspico
remote: Enumerating objects: 446, done.
remote: Counting objects: 100% (392/392), done.
remote: Compressing objects: 100% (129/129), done.
remote: Total 345 (delta 251), reused 300 (delta 209), pack-reused 0 (from 0)
Receiving objects: 100% (345/345), 51.67 KiB | 622.00 KiB/s, done.
Resolving deltas: 100% (251/251), completed with 25 local objects.
From github.com:libsidplayfp/libsidplayfp
 * [new ref]           refs/pull/216/head -> uspico
❯ git merge uspico
Updating 15ce9f75..4e980d9f
Fast-forward
 .github/workflows/build.yml                    |   4 +++-
 .gitmodules                                    |   3 +++
 Makefile.am                                    |  38 +++++++++++++++++++++++++++++++++++---
 configure.ac                                   |  39 +++++++++++++++++++++++++++++++++++++++
 libsidplayfp.pc.in                             |   2 +-
 src/builders/usbsid-builder/driver             |   1 +
 src/builders/usbsid-builder/usbsid-builder.cpp |  62 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 src/builders/usbsid-builder/usbsid-emu.cpp     | 159 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 src/builders/usbsid-builder/usbsid-emu.h       |  83 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 src/builders/usbsid-builder/usbsid.h           |  31 +++++++++++++++++++++++++++++++
 10 files changed, 417 insertions(+), 5 deletions(-)
 create mode 160000 src/builders/usbsid-builder/driver
 create mode 100644 src/builders/usbsid-builder/usbsid-builder.cpp
 create mode 100644 src/builders/usbsid-builder/usbsid-emu.cpp
 create mode 100644 src/builders/usbsid-builder/usbsid-emu.h
 create mode 100644 src/builders/usbsid-builder/usbsid.h
❯ git status
On branch master
Your branch is ahead of 'origin/master' by 65 commits.
  (use "git push" to publish your local commits)

nothing to commit, working tree clean

```

LouDnl commented:
I've noticed an issue when no usbsid is connected but sidplayfp is started with `--usbsid` that causes a segfault. I know where this issue comes from and will commit a fix to this pr tomorrow.

LouDnl commented:
> I've noticed an issue when no usbsid is connected but sidplayfp is started with `--usbsid` that causes a segfault. I know where this issue comes from and will commit a fix to this pr tomorrow.

@drfiemost this should be fixed in [36c1b15](https://github.com/libsidplayfp/libsidplayfp/pull/216/commits/36c1b15790269dfa95325cd6df502d9b70970180)
